pkg/nsx/services/parse.go

```diff
@@ -0,0 +1,43 @@
+package services
+
+import (
+	"errors"
+
+	"github.com/vmware-tanzu/nsx-operator/pkg/apis/v1alpha1"
+	"github.com/vmware-tanzu/nsx-operator/pkg/util"
+)
+
+func getRuleAction(rule *v1alpha1.SecurityPolicyRule) (string, error) {
+	ruleAction := util.ToUpper(*rule.Action)
+	validRuleActions := []string{
```

Comment:
line 12-16 could be put in a var block.
we could also use map or set to implement getRuleAction

Reply:
ok